source/credentials_provider_default_chain.c

```diff
@@ -332,6 +333,17 @@ struct aws_credentials_provider *aws_credentials_provider_new_chain_default(
         aws_atomic_fetch_add(&impl->shutdowns_remaining, 1);
     }
 
+    struct aws_credentials_provider_process_options process_options;
+    AWS_ZERO_STRUCT(process_options);
+    process_options.shutdown_options = sub_provider_shutdown_options;
+    process_options.config_profile_collection_cached = options->profile_collection_cached;
+    process_provider = aws_credentials_provider_new_process(allocator, &options);
+    if (process_provider != NULL) {
+        providers[index++] = process_provider;
+        /* 1 shutdown call from the process provider's shutdown */
+        aws_atomic_fetch_add(&impl->shutdowns_remaining, 1);
+    }
+
     struct aws_credentials_provider_sts_web_identity_options sts_options;
```

Comment:
Add comments to hint that the order is very carefully selected, and some of the reasoning behind that order. Maybe something like:

> Put providers that touch fast local resources above this comment.
> Put providers that do slow networking below this comment.

???